### PR TITLE
support for self-hosted github actions runners

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

File: conda_smithy/templates/github-actions.tmpl [hunk 1]
```diff
@@ -26,11 +34,29 @@ jobs:
             UPLOAD_PACKAGES: {{ data.upload }}
         {%- if data.build_platform.startswith("osx") %}
             os: macos
+        {%- if github_actions.self_hosted %}
+            labels: ['macOS', 'self-hosted', 'x64']
+        {%- endif %}
```
Comment (xhochy) [inline conda_smithy/templates/github-actions.tmpl:39]: We probably can have M1 instances here, too?
```suggestion
        {%- if data.build_platform.startswith("osx-64") %}
            os: macos
        {%- if github_actions.self_hosted %}
            labels: ['macOS', 'self-hosted', 'x64']
        {%- endif %}
        {%- if data.build_platform.startswith("osx-arm64") %}
            os: macos
        {%- if github_actions.self_hosted %}
            labels: ['macOS', 'self-hosted', 'ARM64']
        {%- endif %}
```

File: news/self-hosted-gha.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
+**Added:**
+
+* Added support for self-hosted github actions runners
+
+  In conda-forge.yml, add ``github_actions: self_hosted: true`` to
```
[the remark this answers is not in the document]
Comment (isuruf) [reply news/self-hosted-gha.rst:5]: PRs will not be checked. It'll be too expensive to check PR builds. After merging to master, the builds will be uploaded.
As you said, if a maintainer wants to check, they can push to a branch with uploading disabled. There's also the option of checking that the build works until 6 hours by enabling both Azure and github actions for the same build, but this won't check that the build succeeds fully.

File: conda_smithy/templates/github-actions.tmpl (hunk 1 above)
Comment (isuruf) [reply conda_smithy/templates/github-actions.tmpl:39]: There's no support yet. See https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners#architectures

Comment (chrisburr) [reply conda_smithy/templates/github-actions.tmpl]: Apparently it's possible to use Rosetta 2 to for the runner itself and then escape back to arm64: https://github.com/actions/runner/issues/805

File: conda_smithy/templates/github-actions.tmpl [hunk 3]
```diff
@@ -11,26 +11,57 @@
...
+on: [push]
+{%- else %}
 on: [push, pull_request]
+{%- endif %}
 
 jobs:
   build:
     name: {% raw %}${{ matrix.CONFIG }}{% endraw %}
+    {%- if github_actions.self_hosted %}
+    runs-on {% raw %}${{ matrix.labels }}{% endraw %}
```
Comment (xhochy) [inline conda_smithy/templates/github-actions.tmpl:24]:
```suggestion
    runs-on: {% raw %}${{ matrix.labels }}{% endraw %}
```

**Review by xhochy — APPROVED** [review]

File: conda_smithy/templates/github-actions.tmpl (hunk 1 above)
Comment (isuruf) [reply conda_smithy/templates/github-actions.tmpl]: Great. We might need more changes. Can you open an issue to track?

**Pull request merged**